### PR TITLE
First code contribution: data warehouse loader, readme and license

### PR DESCRIPTION
**Pull request by imrehg**
And project README files, and license as per #4

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -0,0 +1,21 @@
+MIT License
+
+Copyright (c) 2020 NHSX
```
Comment (pbugnion) [inline LICENSE:3]: On [this repository](https://github.com/nhsx/self-isolation-prototype/blob/master/LICENSE), they have used NHSX.UK as the license copyright. We should probably copy that.

File: warehouse-loader/warehouse-loader.py [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+    :param obj: the object in queustion
+    :type key: boto3.resource('s3').ObjectSummary
+    :return: the original object, and a new key where it should be copied within the bucket
+    :rtype: (boto3.resource('s3').ObjectSummary, string)
+    """
+    result = (None, None)
+    tmp = tempfile.NamedTemporaryFile(suffix=".dcm", delete=False)
+    with open(tmp.name, "wb") as fd:
+        obj.Object().download_file(tmp.name)
+    with open(tmp.name, "rb") as fd:
```
Comment (pbugnion) [inline warehouse-loader/warehouse-loader.py:135]: As mentioned, I think the file could be read directly into memory. See e.g.:

https://stackoverflow.com/questions/44043036/how-to-read-image-file-from-s3-bucket-directly-into-memory

**Comment by pbugnion** [comment]
This looks great! Besides the comments I'd made directly, I think this should be merged.

File: LICENSE (hunk 1 above)
Comment (imrehg) [reply LICENSE:3]: Fair, though on this repository it used `NHSX` which I copied (I haven't checked all examples)

https://github.com/nhsx/nhsx-website/blob/master/LICENSE#L3

File: warehouse-loader/warehouse-loader.py (hunk 2 above)
Comment (imrehg) [reply warehouse-loader/warehouse-loader.py:135]: This code is before your feedback last weekend, and haven't added those changes yet. Thanks for the direct link for extra context. Though that you linked is not much different, just fewer lines of code, but it still uses `tempfile.NamedTemporaryFile()`, so both mine and SO both keeps things in memory (on tmpfs).  I have actually tried something very similar, but then `pydicom` was unhappy, hence the extra reopen, but revisiting this next.

File: LICENSE (hunk 1 above)
Comment (pbugnion) [reply LICENSE:3]: ah, then that's fine.

File: warehouse-loader/warehouse-loader.py (hunk 2 above)
Comment (pbugnion) [reply warehouse-loader/warehouse-loader.py:135]: The second answer puts it directly into a StringIO object. The third answer uses BytesIO, which is probably even more appropriate?

File: LICENSE (hunk 1 above)
Comment (AdamChrimes) [reply LICENSE:3]: Yep this should be NHSX 👍

File: warehouse-loader/warehouse-loader.py (hunk 2 above)
Comment (imrehg) [reply warehouse-loader/warehouse-loader.py:135]: Yeah, that's fair, I didn't check that. And indeed we talked about BytesIO, thanks for the patience! :)

**Comment by imrehg** [comment]
@pbugnion updated the file read as you mentioned, and tested it out with dummy data to work.

**Pull request merged**